### PR TITLE
Gh-387: Update the "What is Gaffer" page in the user guide

### LINKED ISSUE #387: Update the "What is Gaffer?" page in the User Guide
It would be good to have a general what is Gaffer page to ensure that the end user / developer is aware of what Gaffer is and how they can use it in their next project.

This could include some diagrams of general interactions and bit of high level background information.

### PR DESCRIPTION
**Pull request by tb06904**
Add a basic page to give an intro as to what is gaffer. 

Includes some high level diagrams and some general features gaffer can offer.

# Related Issue

- Resolve #387

### CODE REVIEW

File: docs/user-guide/gaffer-basics/what-is-gaffer.md [hunk 1]
```diff
@@ -21,4 +22,24 @@ flowchart TD
         B --> C(Gaffer)
         C --> D(key-value store)
     end
-```
\ No newline at end of file
+```
+
+!!! note
+    Knowledge of the [Gaffer schema](../schema.md) in use is usually required to
+    formulate a graph query but technically optional hence the dotted connection.
```
Comment (GCHQDeveloper314) [inline docs/user-guide/gaffer-basics/what-is-gaffer.md]:
```suggestion
    formulate a graph query, but doesn't form part of the query itself - hence the dotted connection.
```

File: docs/user-guide/gaffer-basics/what-is-gaffer.md [hunk 2]
```diff
@@ -21,4 +22,24 @@ flowchart TD
...
+
+## Why would you want to use Gaffer?
+
+Conceptually a graph database/data platform can have a large variety of uses
+where it'll provide specific advantages over other data storage/analytic
+platforms which will be discussed in the ["What is Graph?"](./what-is-a-graph.md)
```
Comment (GCHQDeveloper314) [inline docs/user-guide/gaffer-basics/what-is-gaffer.md]:
```suggestion
platforms which is discussed in the ["What is Graph?"](./what-is-a-graph.md)
```
Even if this isn't completed yet, we don't want to have to come back and change this page I don't think.

File: docs/user-guide/gaffer-basics/what-is-gaffer.md [hunk 3]
```diff
@@ -21,4 +22,24 @@ flowchart TD
...
+section of these docs. Gaffer provides an extensible and straightforward way to
+insert, manage and query the graph data stored in the underlying storage systems
+(ie. Accumulo) that is native to the community.
```
Comment (GCHQDeveloper314) [inline docs/user-guide/gaffer-basics/what-is-gaffer.md]: "native to the community" - I don't understand this

Comment (tb06904) [reply docs/user-guide/gaffer-basics/what-is-gaffer.md]: I think it doesn't really add much to the sentence so will remove

**Review by GCHQDeveloper314 — APPROVED** [review]

**Review by rj77259 — APPROVED** [review]

**Pull request merged**